api/src/API/Controllers/RaceResultsController.cs

```diff
@@ -90,26 +91,30 @@ public async Task<IActionResult> GetOneRaceResult(string orgId, string memberId,
         [HttpPost]
         public async Task<IActionResult> Create(string orgId, string memberId, RaceResult raceResult)
         {
-            if (raceResult.MemberId != Guid.Parse(memberId))
-            {
-                return BadRequest();
-            }
-
-            // Verify member and race exist
             MemberContainerClient memberContainer = containerProvider.MemberContainer;
+
             RaceContainerClient raceContainer = containerProvider.RaceContainer;
 
-            if (!(await memberContainer.ItemExistsAsync(raceResult.MemberId.ToString(), orgId)))
+            // Verify race exists
+            if (!await raceContainer.ItemExistsAsync(raceResult.RaceId.ToString(), raceResult.RaceId.ToString()))
             {
-                return BadRequest($"A member with ID {raceResult.MemberId.ToString()} could not be found");
+                return BadRequest($"A race with ID {raceResult.RaceId} could not be found");
             }
 
-            if (!(await raceContainer.ItemExistsAsync(raceResult.RaceId.ToString(), raceResult.RaceId.ToString())))
+            // Verify member exists
+            if (!await memberContainer.ItemExistsAsync(raceResult.MemberId.ToString(), orgId))
+            {
+                return BadRequest($"A member with ID {raceResult.MemberId} could not be found");
+            }
+
+            var member = await memberContainer.GetOneAsync(raceResult.MemberId.ToString(), orgId);
+
+            // Verify the logged in user is for the member referenced by the POSTed RaceResult
+            if (!await WildApricotController.Authorized(Request, member.OrgAssignedMemberId))
```

Comment:
Do we really need to pass in the whole request?

Reply:
We need two pieces of info that I send via headers. I found it less redundant to pass in the request instead of getting them out each time I call this method.